sql/modules/Util.sql

```diff
@@ -57,6 +57,36 @@ $$ SELECT coalesce((select description FROM language
                    WHERE setting_key = 'default_language')), 'english');
 $$ LANGUAGE sql;
 
+CREATE OR REPLACE FUNCTION language__is_used(in_code text)
+RETURNS BOOLEAN AS $$
+BEGIN
+  PERFORM 1 FROM user_preference
```

Comment:
Ah. ouch. there's no foreign key constraint on that table, is what you're saying. That's not  for the scope of this change, but other than that, it's a bit painful.